README.md

```diff
@@ -27,10 +27,22 @@ This repository provides an interface-level C++ structure for developing an AI c
 
 ```plaintext
 ├── include/                         # Header interfaces for each component
-
+│   ├── IConfig.h
+│   ├── IConversationClient.h
+│   ├── IConversationHistory.h
+│   ├── IMessage.h
+│   ├── INLPProcessor.h
+│   └── IResponseFormatter.h
 ├── src/                             # Empty in HW2, will contain implementation in HW3
```

Comment:
this is a nit, but i would not mention HW2...
make it real.
say something like "interface only for now, implementation to follow in separate PR"